file-attribute-caching/src/test/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingFileSystemTests.kt

```diff
@@ -31,10 +33,52 @@ import java.nio.file.attribute.UserPrincipal
 import java.text.SimpleDateFormat
 import java.util.EnumSet
 import java.util.stream.Stream
+import kotlin.concurrent.thread
 import kotlin.io.path.exists
 
 class AttributeCachingFileSystemTests {
 
+    @Disabled // Enable this test if you wish to see how the attribute caching fileSystem fails in a multi-threaded
+    // environment. This test will be very inconsistent, but it will throw a ClassCastException when it does fail
+    // (you may have to run it multiple times).
+    // The ClassCastException occurs within the map used for the FileSystems stored within the
+    // AttributeCachingFileSystemProvider.
+    @Test
+    fun `attribute caching filesystem is not thread safe`() {
+        var caughtException: Exception? = null
+        val defaultFileSystem = FileSystems.getDefault()
+        val loopCount = 100_000
+        val waitTimeMillis: Long = 50_000
+
+        fun runTestThread(): Thread {
+            return thread {
+                for (i in 1..loopCount) {
+                    try {
+                        AttributeCachingFileSystem.wrapping(defaultFileSystem).use { }
+                    } catch (e: Exception) {
+                        // Test fail fast and verify the exception thrown here
+                        assertThat(e).isInstanceOf(ClassCastException::class.java)
```

Comment:
I'd maybe consider removing this assertion just so that we still have room to catch and allow ConcurrentModificationException, even though we weren't able to reproduce it.

Otherwise I think this is probably fine.

Reply:
gonna keep inside the exception catch itself to fail faster and catch the first instance- rephrased the test name to highlight the specific exception being caught